examples/sampling/graphbolt/quickstart/link_prediction.py

```diff
@@ -93,7 +93,10 @@ def evaluate(model, dataset, device):
         # Forward.
         y = model(data.blocks, x)
         logit = (
-            model.predictor(y[compacted_pairs[0]] * y[compacted_pairs[1]])
+            model.predictor(
+                torch.index_select(y, 0, compacted_pairs[0])
```

Comment:
`index_select()` is not quite straightforward. So I am wondering why torch does not support indexing via int32. And should we follow torch?

Reply:
It is indeed confusing, but currently the only way to avoid using `index_select()` is to convert all variable that used as indices to `torch.int64`.